### ISSUE TITLE
IPlotControl: Extension method for simplified axis linking

### ISSUE THREAD

**Issue by swharden** [ScottPlot/ScottPlot#4003]
Create something like this

```cs
public static class ScottPlotExtensions
{
    /// <summary>
    /// Update another plot control's axis limits when this one's change
    /// </summary>
    public static void ShareAxisLimits(this IPlotControl source, IPlotControl target, bool x = true, bool y = true)
    {
        if (source == target)
            throw new ArgumentException("Cannot link a plot control to itself");

        source.Plot.RenderManager.AxisLimitsChanged += (s, e) =>
        {
            AxisLimits sourceLimits = source.Plot.Axes.GetLimits();
            if (x) target.Plot.Axes.SetLimitsX(sourceLimits.Left, sourceLimits.Right);
            if (y) target.Plot.Axes.SetLimitsY(sourceLimits.Bottom, sourceLimits.Top);
            target.Plot.RenderManager.DisableAxisLimitsChangedEventOnNextRender = true;
            target.Refresh();
        };
    }
}
```

which makes it easy to link plot controls together like this:

```cs
formsPlot1.ShareAxisLimits(formsPlot2, x: true, y: false);
formsPlot2.ShareAxisLimits(formsPlot1, x: true, y: false);
```

Then update the docs to show how to use it

https://scottplot.net/demo/5.0/#shared-axes

**Comment by swharden** [comment]
... note that an "unlink" option should also be added to disable this feature.

Technically this logic could be moved inside the render manager, and plots could store a list of IPlotControl objects to update in this way at the end of a render. That may be cleaner than extension methods on interfaces, and would make it easier to remove links too.

**Comment by swharden** [comment]
I created a new shared axis rule system that allows this API:

```cs
// link specific axes (useful for good for multi-axis plots)
formsPlot2.Plot.Axes.Link(formsPlot2.Plot.Axes.Left, formsPlot3.Plot.Axes.Left, formsPlot3.Plot);

// link axes of two controls together bidirectionally
formsPlot3.Plot.Axes.Link(formsPlot2);
formsPlot3.Plot.Axes.Link(formsPlot2);

// unlink rules for specific plots
formsPlot2.Plot.Axes.Unlink(formsPlot3);
formsPlot3.Plot.Axes.Unlink(formsPlot2);

// unlink all shared axis rules
formsPlot2.Plot.Axes.UnlinkAll();
```

**Comment by swharden** [comment]
Resolved in 6688a24
